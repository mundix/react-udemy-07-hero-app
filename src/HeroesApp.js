import React, { useEffect, useReducer } from 'react'
import { AppRouter } from './routers/AppRouter';
import { AuthContext } from './components/auth/AuthContext';
import { authReducer } from './components/auth/authReducer';

export const HeroesApp = () => {

    const init = () => {
        // Leo si esta logueado por localStorage si tiene l eobjeto , si no manda logged false
        return JSON.parse(localStorage.getItem('user')) || { logged: false };
    }
    //En el componente de mayor relevancia , se llama al reducer paraenviarlo atraves de las useContext custom
    const [user, dispatch] = useReducer(authReducer, {}, init);

    // Guarda el user de la session (En caso de queel token cambien debe reiniciar o enviar a loguear)
    useEffect(() => {
        localStorage.setItem('user', JSON.stringify(user));
    }, [user]);

    return (
        <AuthContext.Provider value={{user,dispatch}}>
            <AppRouter />
        </AuthContext.Provider>
    )
}
