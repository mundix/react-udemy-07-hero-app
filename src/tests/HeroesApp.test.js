import React from 'react';
import { Enzyme, shallow } from 'enzyme';
import { HeroesApp } from '../HeroesApp';


describe('Pruebas en <HeroesApp/>', () => {
    test('Muestra correctamente <HeroesApp/>', () => {
        const wrapp = shallow(<HeroesApp/>);
        expect(wrapp).toMatchSnapshot();
    });
});
