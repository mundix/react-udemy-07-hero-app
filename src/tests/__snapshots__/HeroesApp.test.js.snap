// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroesApp/> Muestra correctamente <HeroesApp/> 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "user": Object {
        "logged": false,
      },
    }
  }
>
  <AppRouter />
</ContextProvider>
`;
